### ISSUE TITLE
Use of native terminal's cursor

### ISSUE THREAD

**Issue by shindere** [anthropics/claude-code#36765]
### Preflight Checklist

- [x] I have searched [existing issues](https://github.com/anthropics/claude-code/issues?q=is%3Aissue%20state%3Aopen%20label%3Abug) and this hasn't been reported yet
- [x] This is a single bug report (please file separate reports for different bugs)
- [x] I am using the latest version of Claude Code

### What's Wrong?

For context, I am running Claude Code under Linux's virtual text console.
I am blind and use BRLTTY to render the content of the screen in braille.
As many screen reading programs, BRLTTY by default uses the terminal's native cursor to decide which portion of the screen to show to the user.
As Claude Code does not use the terminal's native cursor, it is not convenient ot enter prompts as you don't know where your cursor is.
The native cursor remains stuck at the bottom of the screen and, should you want to fix a typo you have no clue where the cursor is.
Ideally, using the terminal's native cursor should be the default but, at the very least there should be a way to configure Claude to do that.
In particular, at the beginning when choosing Claude's configuration, there is no option to make it screen-reader friendly and there would certainly be space for such options, because some features of Claude Code like the use of Unicode characters or the display of ephemeral messages is not so convenient when using a screen reader.

### What Should Happen?

See above.

### Error Messages/Logs

```shell

```

### Steps to Reproduce

Simply run Claude under Linux's virtual console and maybe use BRLTTY.

### Claude Model

None

### Is this a regression?

Yes, this worked in a previous version

### Claude Code Version

2.1.80

### Platform

Anthropic API

### Operating System

Ubuntu/Debian Linux

### Terminal/Shell

Other

**Comment by l3land** [comment]
I am a low-vision Claude pro subscriber and this problem affects me on MacOS and Linux (Debian). Please fix this!

**Comment by PratikP1** [comment]
As a blind Claude Max subscriber, I agree with this issue. I'm encountering the same problem on Windows. I'd like to say that Desktop makes things better. But, it has significant issues as well. Considering how well accessibility and other skills can be incorporated, I'm incredibly surprised and dismayed by the fact that we're dealing with inaccessible interfaces.
